joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/AstCreator.scala

```diff
@@ -1898,7 +1898,7 @@ class AstCreator(filename: String, javaParserAst: CompilationUnit, global: Globa
       // it shouldn't cause issues since resolvedType is only used where the extra type
       // information not available in typeName is necessary.
       val typeName     = expressionReturnTypeFullName(value)
-      val resolvedType = Try(value.calculateResolvedType()).toOption
+      val resolvedType = tryWithSafeStackOverflow(value.calculateResolvedType()).toOption
```

Comment:
I found these new stack overflows while working on this change, but it seems unrelated (and is rather a consequence of switching to the JdkJarTypeSolver). Figuring out why these occur and fixing them is a TODO, but I included the patch in here as a temporary remedy.